subsys/nrf_security/src/drivers/cracen/cracenpsa/src/kmu.c

```diff
@@ -543,17 +562,22 @@ static psa_status_t convert_to_psa_attributes(kmu_metadata *metadata,
 				: PSA_KEY_TYPE_ECC_PUBLIC_KEY(PSA_ECC_FAMILY_TWISTED_EDWARDS));
 		psa_set_key_algorithm(key_attr, PSA_ALG_ED25519PH);
 		break;
+#endif
+#ifdef PSA_WANT_ALG_ECDSA
 	case METADATA_ALG_ECDSA:
 		psa_set_key_type(key_attr,
 				 can_sign(key_attr)
 					 ? PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1)
 					 : PSA_KEY_TYPE_ECC_PUBLIC_KEY(PSA_ECC_FAMILY_SECP_R1));
 		psa_set_key_algorithm(key_attr, PSA_ALG_ECDSA(PSA_ALG_ANY_HASH));
 		break;
+#endif
+#ifdef PSA_WANT_ALG_HMAC
 	case METADATA_ALG_HMAC:
 		psa_set_key_type(key_attr, PSA_KEY_TYPE_HMAC);
 		psa_set_key_algorithm(key_attr, PSA_ALG_HMAC(PSA_ALG_SHA_256));
 		break;
+#endif
 	default:
 		return PSA_ERROR_HARDWARE_FAILURE;
```

Comment:
The thing is that now if a PSA_WANT is not defined it will return PSA_ERROR_HARDWARE_FAILURE which is a bit strange, no?

I think that returning a PSA_ERROR_NOT_SUPPORTED is more consistent to what we do in other parts of the driver.

Reply:
I can change it, but  I think it makes sense given the circumstances. This is called on when you convert metadata from a key stored in the KMU back to a PSA key right? With these new settings you should not be able to store a key in the KMU that you don't have the config for, so if you get to this default then some corruption should have happened to the metadata. 

I agree it looks bad on its own, but since `convert_from_psa_attributes` should always be call before, which gives the `PSA_ERROR_NOT_SUPPORTED` then once you get here only a hardware failure should lead to this result no?